### PR TITLE
Storing Courses from Google Classroom in the Datastore

### PR DESCRIPTION
**Pull request by rsampdev**

### CODE REVIEW

**Review by BenLin00 — DISMISSED** [review]

**Review by mahimashanware — COMMENTED** [review]
target files

File: project/src/main/java/com/google/autograder/servlets/ListCoursesServlet.java [hunk 1]
```diff
@@ -54,4 +86,81 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response) t
...
+
+                Services.DATA_STORE.put(courseEntity);            
+            } 
+
+        } catch (Exception e) {
+
+            e.printStackTrace();
+        
+        } finally {
+
```
Comment (BenLin00) [inline project/src/main/java/com/google/autograder/servlets/ListCoursesServlet.java:131]: is this needed?

File: project/src/main/java/com/google/autograder/servlets/ListCoursesServlet.java [hunk 2]
```diff
@@ -54,4 +86,81 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response) t
...
+    private void storeCourseDataInDatastore(String courseJSON) {
+        try {
+            String userEmail = Services.USER_SERVICE.getCurrentUser().getEmail();
+            String userID = Services.USER_SERVICE.getCurrentUser().getUserId();
+
+            Filter userEmailFilter = new FilterPredicate("userEmail", FilterOperator.EQUAL, userEmail);
+            Query query = new Query("Course").setFilter(userEmailFilter);
+            PreparedQuery results = Services.DATA_STORE.prepare(query);
+
+            for (Entity course : results.asIterable()) {
```
Comment (ecao1107) [inline project/src/main/java/com/google/autograder/servlets/ListCoursesServlet.java:98]: Why do you need to clean up all courses before storing?

File: project/src/main/java/com/google/autograder/servlets/ListCoursesServlet.java [hunk 3]
```diff
@@ -44,7 +64,19 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response) t
...
                 String json = API.getJSON(connection);
 
-                response.getWriter().println(json);
+                // System.out.println(json);
+
+                // System.out.println("\n\n\n");
+
+                storeCourseDataInDatastore(json);
+
+                String coursesData = getCoursesDataFromDatastore();
```
Comment (ecao1107) [inline project/src/main/java/com/google/autograder/servlets/ListCoursesServlet.java:73]: An improvement would be:
1. ListCourses from Classroom api
2. GetCourses data from data store.
3. Compare the result, if different, then call storeCourseDataInDatastore.

You can add a follow up PR for this improvement.

Comment (rsampdev) [reply project/src/main/java/com/google/autograder/servlets/ListCoursesServlet.java:73]: Yeah that makes sense, I'm going to get to that soon. I wanted to get the flow working from Course -> Assignment -> Submissions working first before I optimized it. I also wanted to wait until after I integrated with Mahima's larger Database class.

File: project/src/main/java/com/google/autograder/servlets/ListCoursesServlet.java (hunk 1 above)
Comment (rsampdev) [reply project/src/main/java/com/google/autograder/servlets/ListCoursesServlet.java:131]: No, you're right, this finally statement  is not needed. I removed it in the PR right after this I believe.

**Review by mahimashanware — APPROVED** [review]

**Review by BenLin00 — APPROVED** [review]
lgtm

**Pull request merged**